src/lib/components/wizard.component.ts

```diff
@@ -279,8 +249,8 @@ export class WizardComponent implements AfterContentInit, OnChanges {
    *
    * @param navigationMode The updated navigation mode
    */
-  public updateNavigationMode(navigationMode: NavigationModeInput) {
-    this._navigation = this.navigationModeFactory.create(this, navigationMode);
+  public updateNavigationMode(navigationMode: NavigationMode): void {
```

Comment:
This method looks to me as if it should be renamed to:
```typescript
public set navigationMode(navigationMode: NavigationMode) { ... }
```

Slightly off-topic I think the same should be true for:

```typescript
 /**
   * Updates the wizard steps to the new array
   *
   * @param wizardSteps The updated wizard steps
   */
  private updateWizardSteps(wizardSteps: WizardStep[]): void {
    // the wizard is currently not in the initialization phase
    if (this.wizardSteps.length > 0 && this.currentStepIndex > -1) {
      this.currentStepIndex = wizardSteps.indexOf(this.wizardSteps[this.currentStepIndex]);
    }

    this._wizardSteps = wizardSteps;
}
```

Reply:
I think it would be better to change `updateNavigationMode` into `set navigation` (not `set navigationMode`), because there is already

```
public get navigation(): NavigationMode {
  return this._navigation;
}
```

Reply:
Yes, sorry I accidentally wrote `navigationMode` while I meant: `navigation`.

In my opinion `navigation` is more on point for an user of the library because the field is used to perform "navigation" on the wizard. `navigationMode` is more abstract in comparison and more intuitive. From a library developer perspective you're right, that `navigationMode` is a better choice because it better describes the type of object that can be assigned to the field.

I'm fine either way (i.e. `navigationMode` or `navigation`). We just need to be aware that we shouldn't change the name later on because the current changes are meant to create a stable API, which can be used by programmers to access the navigation functionality of the wizard from outside the library.

Reply:
Changed into `set navigation` in 5e67153.

`updateWizardSteps` cannot be changes the same way. The `updateWizardSteps` method is private, but the `wizardSteps` getter is public. In TypeScript, getter and setter for the same field are required to have the same visibility (for some reason). So, we are out of luck unless we are ready to make the setter public (which I think we aren't).